extensions/blocks/donations/view.js

```diff
@@ -20,226 +20,225 @@ import { initializeMembershipButtons } from '../../shared/memberships';
+		this.amount = null;
+		this.isCustomAmount = false;
+		this.interval = 'one-time';
+
+		// Initialize block.
+		this.initNavigation();
+		this.handleCustomAmount();
+		this.handleChosenAmount();
+
+		// Remove loading spinner.
+		this.block.querySelector( '.donations__container' ).classList.add( 'loaded' );
 	}
-};
-
-const getDonateButton = interval => {
-	const buttonIntervalClasses = {
-		'one-time': 'donations__one-time-item',
-		'1 month': 'donations__monthly-item',
-		'1 year': 'donations__annual-item',
-	};
-	return document.querySelector(
-		`.wp-block-jetpack-donations .donations__donate-button.${ buttonIntervalClasses[ interval ] }`
-	);
-};
-
-const toggleDonateButton = enable => {
-	const donateButton = getDonateButton( jetpackDonationsInterval );
-	if ( enable ) {
-		donateButton.classList.remove( 'is-disabled' );
-	} else {
-		donateButton.classList.add( 'is-disabled' );
+
+	getNavItem( interval ) {
+		return this.block.querySelector( `.donations__nav-item[data-interval="${ interval }"]` );
 	}
-};
-
-const updateUrl = () => {
-	const donateButton = getDonateButton( jetpackDonationsInterval );
-	const url = donateButton.getAttribute( 'href' );
-	if ( jetpackDonationsAmount ) {
-		donateButton.setAttribute(
-			'href',
-			addQueryArgs( url, {
-				amount: jetpackDonationsAmount,
-				...( jetpackDonationsIsCustomAmount && { customAmount: true } ),
-			} )
-		);
-	} else {
-		donateButton.setAttribute( 'href', removeQueryArgs( url, 'amount', 'customAmount' ) );
+
+	resetSelectedAmount() {
+		const selectedAmount = this.block.querySelector( '.donations__amount.is-selected' );
+		if ( selectedAmount ) {
+			selectedAmount.classList.remove( 'is-selected' );
+		}
 	}
-};
-
-const updateAmountFromCustomAmountInput = () => {
-	const input = document.querySelector(
-		'.wp-block-jetpack-donations .donations__custom-amount .donations__amount-value'
-	);
-	const wrapper = document.querySelector( '.wp-block-jetpack-donations .donations__custom-amount' );
-
-	const amount = input.innerHTML;
-	if ( ! amount ) {
-		jetpackDonationsAmount = null;
-		toggleDonateButton( false );
-		return;
+
+	getDonateButton() {
+		const buttonIntervalClasses = {
+			'one-time': 'donations__one-time-item',
+			'1 month': 'donations__monthly-item',
+			'1 year': 'donations__annual-item',
+		};
+		return this.block.querySelector(
+			`.donations__donate-button.${ buttonIntervalClasses[ this.interval ] }`
+		);
 	}
 
-	// Validates the amount.
-	const currency = input.dataset.currency;
-	const parsedAmount = parseAmount( amount, currency );
-	if ( parsedAmount && parsedAmount >= minimumTransactionAmountForCurrency( currency ) ) {
-		wrapper.classList.remove( 'has-error' );
-		jetpackDonationsAmount = parsedAmount;
-		toggleDonateButton( true );
-	} else {
-		wrapper.classList.add( 'has-error' );
-		jetpackDonationsAmount = null;
-		toggleDonateButton( false );
+	toggleDonateButton( enable ) {
+		const donateButton = this.getDonateButton();
+		if ( enable ) {
```

Comment:
We can use a ternary here, but not sure if that's the code style we follow -- also, that's just a stylistic opinion :)